### ISSUE TITLE
[BUG][iOS] Calls to UIKit method on a background queue and crashes

### ISSUE THREAD

**Issue by artyom-stv** [microsoft/clarity-apps#172]
**Describe the Bug**
Clarity SDK calls UIKit [accessibilityLabel](https://developer.apple.com/documentation/uikit/uiaccessibilityelement/accessibilitylabel) property on a backgorund queue `displayListQueue`. It leads to crashes (verified in 3.0.17) and can also lead to unpredictable behavior, UI glitches and race conditions.

UIKit documentation strictly states the following:
>Use UIKit classes only from your app’s main thread or main dispatch queue, unless otherwise indicated in the documentation for those classes. This restriction particularly applies to classes that derive from [UIResponder](https://developer.apple.com/documentation/uikit/uiresponder) or that involve manipulating your app’s user interface in any way.

I cannot find notices in the documentation that [accessibilityLabel](https://developer.apple.com/documentation/uikit/uiaccessibilityelement/accessibilitylabel) property can be called from a background queue.

Crash stack trace (queue `displayListQueue`, message: `Modifications to the layout engine must not be performed from a background thread after it has been accessed from the main thread.`):
```
#0	0x0000000105d38874 in __pthread_kill ()
#1	0x00000001050be2ec in pthread_kill ()
#2	0x00000001801704ec in abort ()
#3	0x00000001802ad100 in abort_message ()
#4	0x000000018029cb04 in demangling_terminate_handler ()
#5	0x000000018006f8ac in _objc_terminate ()
#6	0x00000001802ac4d8 in std::__terminate ()
#7	0x00000001802ac488 in std::terminate ()
#8	0x0000000180092f50 in objc_terminate ()
#9	0x0000000104fa65e4 in _dispatch_client_callout ()
#10	0x0000000104fae8b8 in _dispatch_lane_serial_drain ()
#11	0x0000000104faf628 in _dispatch_lane_invoke ()
#12	0x0000000104fbc2cc in _dispatch_root_queue_drain_deferred_wlh ()
#13	0x0000000104fbb838 in _dispatch_workloop_worker_thread ()
#14	0x00000001050babcc in _pthread_wqthread ()
Exception Backtrace#0	0x00000001804b9100 in __exceptionPreprocess ()
#1	0x0000000180092da8 in objc_exception_throw ()
#2	0x00000001dcd6f254 in _AssertAutoLayoutOnAllowedThreadsOnly ()
#3	0x00000001dcd6f63c in -[NSISEngine withBehaviors:performModifications:] ()
#4	0x000000018609c568 in -[UIView(Hierarchy) layoutBelowIfNeeded] ()
#5	0x000000018568b1d4 in -[UIButtonLegacyVisualProvider imageViewCreateIfNeeded:] ()
#6	0x0000000185407b84 in -[UIButton imageView] ()
#7	0x00000001cbf9e7e0 in -[UIButtonAccessibility accessibilityLabel] ()
#8	0x00000001ccb03f28 in -[UIButtonAccessibility__MapKit__UIKit accessibilityLabel] ()
#9	0x00000001cc15bd08 in -[UIButtonAccessibility__Social__UIKit accessibilityLabel] ()
#10	0x00000001cc1dae8c in -[UIButtonAccessibility__PhotosUI__UIKit accessibilityLabel] ()
#11	0x00000001cc7e08c8 in -[UIButtonAccessibility__AnnotationKit__UIKit accessibilityLabel] ()
#12	0x00000001cc09efc8 in -[_UIButtonBarButtonAccessibility accessibilityLabel] ()
#13	0x0000000107afe59c in static ClickProcessor.CandidateTextNode.nodeText(_:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:126
#14	0x0000000107afe270 in static ClickProcessor.CandidateTextNode.create(node:level:clickX:clickY:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:112
#15	0x0000000107afe094 in ClickProcessor.retrieveCandidateTextNodes(_:level:x:y:candidates:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:77
#16	0x0000000107afdcf0 in ClickProcessor.resolveClickText(candidateClickedNode:x:y:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:63
#17	0x0000000107afd3d4 in ClickProcessor.process(_:viewHierarchy:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:10
#18	0x0000000107b2cd18 in closure #1 in CaptureManager.onUserInteraction(_:) at /Users/runner/work/1/s/Clarity/Clarity/Managers/CaptureManager.swift:384
#19	0x00000001078ae6c0 in thunk for @escaping @callee_guaranteed @Sendable () -> () ()
#20	0x0000000104fa4cd8 in _dispatch_call_block_and_release ()
#21	0x0000000104fa65d0 in _dispatch_client_callout ()
#22	0x0000000104fae8b8 in _dispatch_lane_serial_drain ()
#23	0x0000000104faf628 in _dispatch_lane_invoke ()
#24	0x0000000104fbc2cc in _dispatch_root_queue_drain_deferred_wlh ()
#25	0x0000000104fbb838 in _dispatch_workloop_worker_thread ()
#26	0x00000001050babcc in _pthread_wqthread ()
#27	0x00000001050b998c in start_wqthread ()
```

Another example of a stack trace where it is clear that `accessibilityLabel` getter can perform very complex tasks, and calling `accessibilityLabel` on a background queue can lead to unpredictable behavior:
```
#2	0x00000001860a70fc in -[UIView(Internal) _addSubview:positioned:relativeTo:] ()
#3	0x00000001856919a4 in -[UIButtonLegacyVisualProvider _setupImageView] ()
#4	0x000000018568b1bc in -[UIButtonLegacyVisualProvider imageViewCreateIfNeeded:] ()
#5	0x0000000185407b84 in -[UIButton imageView] ()
#6	0x00000001cbf9e7e0 in -[UIButtonAccessibility accessibilityLabel] ()
#7	0x00000001ccb03f28 in -[UIButtonAccessibility__MapKit__UIKit accessibilityLabel] ()
#8	0x00000001cc15bd08 in -[UIButtonAccessibility__Social__UIKit accessibilityLabel] ()
#9	0x00000001cc1dae8c in -[UIButtonAccessibility__PhotosUI__UIKit accessibilityLabel] ()
#10	0x00000001cc7e08c8 in -[UIButtonAccessibility__AnnotationKit__UIKit accessibilityLabel] ()
#11	0x00000001cc09efc8 in -[_UIButtonBarButtonAccessibility accessibilityLabel] ()
#12	0x0000000107afe59c in static ClickProcessor.CandidateTextNode.nodeText(_:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:126
#13	0x0000000107afe270 in static ClickProcessor.CandidateTextNode.create(node:level:clickX:clickY:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:112
#14	0x0000000107afe094 in ClickProcessor.retrieveCandidateTextNodes(_:level:x:y:candidates:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:77
#15	0x0000000107afdcf0 in ClickProcessor.resolveClickText(candidateClickedNode:x:y:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:63
#16	0x0000000107afd3d4 in ClickProcessor.process(_:viewHierarchy:) at /Users/runner/work/1/s/Clarity/Clarity/Helpers/ClickProcessor.swift:10
#17	0x0000000107b2cd18 in closure #1 in CaptureManager.onUserInteraction(_:) at /Users/runner/work/1/s/Clarity/Clarity/Managers/CaptureManager.swift:384
#18	0x00000001078ae6c0 in thunk for @escaping @callee_guaranteed @Sendable () -> () ()
#19	0x0000000104fa4cd8 in _dispatch_call_block_and_release ()
#20	0x0000000104fa65d0 in _dispatch_client_callout ()
#21	0x0000000104fae8b8 in _dispatch_lane_serial_drain ()
#22	0x0000000104faf628 in _dispatch_lane_invoke ()
#23	0x0000000104fbc2cc in _dispatch_root_queue_drain_deferred_wlh ()
#24	0x0000000104fbb838 in _dispatch_workloop_worker_thread ()
#25	0x00000001050babcc in _pthread_wqthread ()
```

**Repro Steps**
Normal app usage.

**Expected Behavior**
Clarity SDK shouldn't call UIKit methods on a background queue if it isn't explicitly allowed in the documentation.

**Screenshots**
-

**Project Info**

- Project ID: shared with clarity-apps-support@microsoft.com

**Platform/Framework Info**

- Device: iPhone 16 Pro Simulator
- OS: iOS 18.3.1
- Application Framework: Native UIKit-based
- Clarity Version 3.0.17

**Additional Context**
-

**Comment by amralaa-MSFT** [comment]
Just to clarify, all references to`accessibilityLabel` were removed in `3.0.17`.

**Comment by artyom-stv** [comment]
Sorry for the inconvenience. It was our fault during update from 3.0.16 to 3.0.17.
